### PR TITLE
codec(ticdc): simple support large message handle functionality

### LINKED ISSUE #10247: support handle key only mode

### LINKED ISSUE #10248: support claim check mode

### LINKED ISSUE #10249: support TiCDC level compression

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #10249, close #10248, close #10247

### What is changed and how it works?

* simple protocol support large message handle functionality, include message level compression, handle key only if the message is too large, and claim check, which send message to the external storage.
* revert DDL manager send bootstrap message, since the latest design is let the processor sending bootstrap message lazily, this will be implemented later.
* add a new unit test to cover the changefeed state nil

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
`None`
```

### CODE REVIEW

**Review by asddongmen — APPROVED** [review]

File: cdc/owner/ddl_sink.go [hunk 1]
```diff
@@ -54,7 +54,6 @@ type DDLSink interface {
 	// the DDL event will be sent to another goroutine and execute to downstream
 	// the caller of this function can call again and again until a true returned
 	emitDDLEvent(ctx context.Context, ddl *model.DDLEvent) (bool, error)
-	emitBootstrapEvent(ctx context.Context, ddl *model.DDLEvent) error
```
Comment (CharlesCheung96) [inline cdc/owner/ddl_sink.go:57]: Why remove this method?

**Comment by CharlesCheung96** [comment]
Please explain in the pr body why some codes, which is not related to this feature, was removed.

File: cdc/owner/ddl_sink.go (hunk 1 above)
Comment (3AceShowHand) [reply cdc/owner/ddl_sink.go:57]: It's removed, since the bootstrap event is going to be sent by the processor.

**Review by CharlesCheung96 — APPROVED** [review]
